mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -1,409 +1,97 @@
-            # records twice
-            if not trainer.enable_validation:
-                self._log_epoch_metrics(trainer, pl_module)
-
-        @rank_zero_only
-        def on_validation_epoch_end(self, trainer, pl_module):
-            """
-            Log loss and other metrics values after each validation epoch
-
-            :param trainer: pytorch lightning trainer instance
-            :param pl_module: pytorch lightning base module
-            """
-            self._log_epoch_metrics(trainer, pl_module)
-
-    else:
-
-        @rank_zero_only
-        def on_epoch_end(self, trainer, pl_module):
-            """
-            Log loss and other metrics values after each epoch
-
-            :param trainer: pytorch lightning trainer instance
-            :param pl_module: pytorch lightning base module
-            """
-            self._log_epoch_metrics(trainer, pl_module)
-
-    @rank_zero_only
-    def on_train_batch_end(
-        self, trainer, pl_module, *args
-    ):  # pylint: disable=signature-differs,arguments-differ,unused-argument
-        """
-        Log metric values after each step
-
-        :param trainer: pytorch lightning trainer instance
-        :param pl_module: pytorch lightning base module
-        """
-        if not self.log_every_n_step:
-            return
-        # When logging at the end of a batch step, we only want to log metrics that are logged
-        # on steps. For forked metrics (metrics logged on both steps and epochs), we exclude the
-        # metric with the non-forked name (eg. "loss" when we have "loss", "loss_step" and
-        # "loss_epoch") so that this is only logged on epochs. We also record which metrics
-        # we've logged per step, so we can later exclude these from metrics logged on epochs.
-        metrics = _get_step_metrics(trainer)
-        metric_items = [
-            (name, val)
-            for (name, val) in metrics.items()
-            if (name not in self._epoch_metrics) and (f"{name}_step" not in metrics.keys())
+    met = Metric(key=key, value=value, timestamp=time, step=step)
+    with _metric_queue_lock:
+        _metric_queue.append((run_id, met))
+        if len(_metric_queue) > _MAX_METRIC_QUEUE_SIZE:
+            _thread_pool.submit(_flush_queue)
+
+
+FAILED = object()
+autolog_run = None
+
+
+def _get_run():
+    global autolog_run
+    if autolog_run is None:
+        autolog_run = mlflow.active_run()
+    if autolog_run is None:
+        try:
+            autolog_run = mlflow.start_run()
+        except Exception:
+            # don't try again.
+            autolog_run = FAILED
+    if autolog_run not in (None, FAILED):
+        return autolog_run
+    return None
+
+
+def patched_add_hparams(original, self, hparam_dict, metric_dict, *args, **kwargs):
+    """use a synchronous call here since this is going to get called very infrequently."""
+
+    run = _get_run()
+
+    if not pl_autolog.IN_FIT and run is not None and hparam_dict:
```

Comment:
Do we need this global variable `IN_FIT`? Can we do something like this?

```
try:
    import pytorch_lightning
    IS_PYTORCH_LIGHTNING_AVAILABLE = True
except ImportError:
    IS_PYTORCH_LIGHTNING_AVAILABLE = False

...

    if not IS_PYTORCH_LIGHTNING_AVAILABLE and run is not None and hparam_dict:
```

Reply:
Even though lightning may be installed in the image, the user doesn't need to use it.

Reply:
I see. Can we disable vanilla pytorch autologging in `patched_fit`?

Reply:
that's what the flag is doing, right? Or what mechanism  did you have in mind?

Reply:
i reversed the logic, lmk if that's what you had in mind.